src/statsmodels/formula.jl

```diff
@@ -481,7 +481,16 @@ creating the model matrix.
         append!(assign, fill(i_term, size(blocks[end], 2)))
     end
 
-    ModelMatrix{T}(reduce(hcat, blocks), assign)
+    I = size(dfrm, 1)
+    J = mapreduce(x -> size(x, 2), +, blocks)
+    X = zeros(eltype(T), I, J)
```

Comment:
@GordStephen Is this OK as regards your recent work on sparse matrices? Or should we use `similar` here?